storage/innobase/page/page0cur.cc

```diff
@@ -1742,7 +1742,13 @@ rec_t *page_cur_insert_rec_zip(
       }
 
       /* Out of space: restore the page */
-      if (!page_zip_decompress(page_zip, page, FALSE)) {
+      const uint32_t encryption_key_version_in =
+          mach_read_from_4(page_zip->data + FIL_PAGE_ENCRYPTION_KEY_VERSION);
+      const uint32_t encryption_key_version_out =
+          mach_read_from_4(page + FIL_PAGE_ENCRYPTION_KEY_VERSION);
```

Comment:
@hagrid-the-developer This feature (keyring key encryption)  is currently unused in PS and so it should never be changed.

Please find out how the KEY_VERSION changed when the feature is disabled?